sql/pgwire/encoding.go

```diff
@@ -78,7 +59,11 @@ func (b *readBuffer) readUntypedMsg(rd io.Reader) error {
 			size, maxMessageSize)
 	}
 
-	b.reset(size)
+	if len(b.msgBuf) < size {
```

Comment:
Yeah, I wondered about that too. It's not obviously safe, but this method seems to be called just once per incoming "request", which would make this OK.